### PR TITLE
Adding a method to serialize Params and FrozenParams objects

### PR DESCRIPTION
**Pull request by minhtuev**
In order to serialize the Request object, we need to serialize Params and Frozen Params

### CODE REVIEW

File: mindmeld/components/request.py [hunk 1]
```diff
@@ -246,6 +246,32 @@ def nlp_params(self):
...
+        fields = [
+            "allowed_intents",
+            "target_dialogue_state",
+            "time_zone",
+            "timestamp",
+            "language",
+            "locale",
+            "dynamic_resource",
+        ]
+        _dic = self.nlp_params()
```
Comment (vijay120) [inline mindmeld/components/request.py:259]: Its unclear why you are using the `nlp_params ` here. The `nlp_params ` method is only used for validation of certain params attributes. Why don't we just iterates through all the object attributes and return them in a dictionary?

Comment (minhtuev) [reply mindmeld/components/request.py:259]: I see, I thought that the `nlp_params` is preferred method to retrieve nlp params, so I wanted to reuse that. I want to add this function to support serialization for both Params and FrozenParams object.

File: mindmeld/components/dialogue.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ def __init__(self, dialogue_state, **kwargs):
         resolved = {}
         for keys in key_kwargs:
             if len(keys) == 2:
-                single, plural = keys
+                single, plural = keys  # pylint: disable=unbalanced-tuple-unpacking
```
Comment (serapio) [inline mindmeld/components/dialogue.py:123]: I realize this is incidental to your changes, but I hadn't realized we were doing this. In L130 we are assuming `single` is a string, and L132 assumes `plural` is an iterable of strings, but the doc string says `has_entity` and `has_entities` are both `(str, list, set)`

File: mindmeld/components/request.py [hunk 3]
```diff
@@ -228,9 +228,10 @@ def dm_params(self, handler_map):
             return {"target_dialogue_state": None}
         return {"target_dialogue_state": target_dialogue_state}
 
-    def nlp_params(self):
+    def validate_nlp_params(self):
         """
-        Validate time zone, timestamp, and dynamic resource parameters.
+        Validate time zone, timestamp, and dynamic resource parameters and return the params as a
```
Comment (serapio) [inline mindmeld/components/request.py:233]: ... plus language and locale. Seems there should be a better conceptual description for the parameters grouped together here. And we should keep the method above with a parallel name (=> validate_dm_params)

File: mindmeld/components/dialogue.py (hunk 2 above)
Comment (minhtuev) [reply mindmeld/components/dialogue.py]: Ah, good eyes! Update docstring.

File: mindmeld/components/request.py (hunk 3 above)
Comment (minhtuev) [reply mindmeld/components/request.py]: @vijay120 : what are the reasons for grouping these together?

File: mindmeld/components/dialogue.py [hunk 4]
```diff
@@ -90,8 +90,8 @@ def __init__(self, dialogue_state, **kwargs):
         Args:
             dialogue_state (str): The name of the dialogue state.
             domain (str): The name of the domain to match against.
-            has_entity (str, list, set): A synonym for the ``has_entities`` param.
-            has_entities (str, list, set): A single entity type or a list of entity types to match
+            has_entity (str): A single entity type to match.
+            has_entities (str, list, set): A single entity type or a list/set of entity types to match
```
Comment (serapio) [inline mindmeld/components/dialogue.py:94]: has_entities shouldn't be a string, because it will turn the string into a set of characters

Comment (minhtuev) [reply mindmeld/components/dialogue.py]: Done

**Pull request merged**